src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java

```diff
@@ -126,15 +126,17 @@ public String getGenericAwareResponseTypeName(TypeRef outputType) {
   public String getAndSavePagedResponseTypeName(
       Method method, ModelTypeTable typeTable, Field resourceField) {
     // TODO(michaelbausor) make sure this uses the typeTable correctly
-    return getPagedResponseWrappersClassName()
-        + "."
-        + getAndSavePagedResponseTypeInnerName(method, typeTable, resourceField);
+
+    String fullPackageWrapperName = typeTable.getFullNameFor(getPagedResponseWrappersClassName());
+    return typeTable.getAndSaveNicknameForStaticInnerClass(
+        fullPackageWrapperName
+            + "."
+            + getPagedResponseTypeInnerName(method, typeTable, resourceField));
```

Comment:
I would prefer getAndSaveNicknameForStaticInnerClass to take two parameters:

1. Container type
2. Inner type

and rename it to getAndSaveNicknameForInnerType.